### PR TITLE
Use rpaths & soname/install_name to link sibling shared libraries on OS X and Linux

### PR DESCRIPTION
**Pull request by tritao**
Original patch by Renaud Guillard (@noresources) updated to recent Premake.

### CODE REVIEW

**Comment by tvandijck** [comment]
It looks good to me.... should I close the other PR and merge this one then?

**Comment by tritao** [comment]
Sounds good to me.

File: src/actions/make/_make.lua [hunk 1]
```diff
@@ -178,9 +178,21 @@
 -- Format a list of values to be safely written as part of a variable assignment.
 --
 
-	function make.list(value)
+	function make.list(value, quoted)
+		quoted = false
```
Comment (noresources) [inline src/actions/make/_make.lua]: Is that the line was intentionally kept or is it a test relict ? If your intention was to suppress the new behavior, you should just have to revert make.list as it was before. No ?

**Comment by tritao** [comment]
@tvandijck mind removing that line from master?

File: src/actions/make/_make.lua (hunk 1 above)
Comment (tritao) [inline src/actions/make/_make.lua]: My mistake, I was testing your PR locally and checking if this change made any difference in the test suite and forgot to remove it the debug line.

@tvandijck mind removing that line from master?

**Pull request merged**